src/studio/ui/StudioPanel.java

```diff
@@ -741,8 +796,14 @@ else if ((row == 1) && ((noWins % 3) > 1))
         }
     }
 
-    private void setServer(Server server) {
-        if (server == null) return;
+    private void setServer(Server server, boolean addToNavigationHistory) {
```

Comment:
Here is what I spot but think a lot how I would do it here.. 
I don't like the boolean addToNavigationHistory argument. As we have different use-cases - replace the current item, add to the tail, move back and forward.
May be we need to have setServer(server) meaning we have a new Server which we want to set and add to the tail; and a separate method navigate(boolean moveForward) which is used for navigation?